src/machine/i2c.go

```diff
@@ -6,6 +6,15 @@ import (
 	"errors"
 )
 
+// i2cController interface defines what is required to provide a i2c controller.
+// It is primarily to provide compile time checking when adding a new i2c
+// implementation.
+type i2cController interface {
```

Comment:
I'd rather we not implement compilation check types in the global namespace if we can avoid it. To do this we can replace this definition with the following in this file to ensure all future implementations fulfill the interface so that we don't have to add the check on each individual interface facilitating maintenance in the long run. We also keep the global namespace free of clutter going forward:
```go
// If you are getting a compile error on this line please check to see you've
// correctly implemented the methods on the I2C type. They must match
// the i2cController interface method signatures type to type perfectly.
// If not implementing the I2C type please remove your target from the build tags at the top of this file.
var _ interface{  // 2
	Configure(config I2CConfig) error
	Tx(addr uint16, w, r []byte) error
	SetBaudRate(br uint32) error
} = (*I2C)(nil)
```

Or 

```go
func _() {
    type i2cController interface {
	    Configure(config I2CConfig) error
	    Tx(addr uint16, w, r []byte) error
	    SetBaudRate(br uint32) error
    }
    // If you are getting a compile error on this line please check to see you've
    // correctly implemented the methods on the I2C type. They must match
    // the i2cController interface method signatures type to type perfectly.
    // If not implementing the I2C type please remove your target from the build tags at the top of this file.
    var _ i2cController = (*I2C)(nil)
}
```

Reply:
See https://github.com/tinygo-org/tinygo/issues/2334#issuecomment-1238872628

Reply:
Good idea, done.